From c5d4234a1e8815a370c58c4e7564fd9c15686d39 Mon Sep 17 00:00:00 2001
From: Dongjin Kim <tobetter@gmail.com>
Date: Wed, 14 Apr 2021 18:00:07 +0900
Subject: [PATCH 68/99] ODROID-N2/C4/HC4: Revert "drm: meson_drv add shutdown
 function"

Reboot fails 100% with this commit.

This reverts commit b0fe7f2c002adecda5f146bc5e430d0e3965c255.

Change-Id: Id91b44dad4359f137dfea2738e3a7705b211e67a
---
 drivers/gpu/drm/meson/meson_drv.c | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/drivers/gpu/drm/meson/meson_drv.c b/drivers/gpu/drm/meson/meson_drv.c
index bd4ca11d3ff5..7d8b2c64dd31 100644
--- a/drivers/gpu/drm/meson/meson_drv.c
+++ b/drivers/gpu/drm/meson/meson_drv.c
@@ -428,15 +428,12 @@ static int __maybe_unused meson_drv_pm_resume(struct device *dev)
 	return drm_mode_config_helper_resume(priv->drm);
 }
 
-static void meson_drv_shutdown(struct platform_device *pdev)
+static int compare_of(struct device *dev, void *data)
 {
-	struct meson_drm *priv = dev_get_drvdata(&pdev->dev);
+	DRM_DEBUG_DRIVER("Comparing of node %pOF with %pOF\n",
+			 dev->of_node, data);
 
-	if (!priv)
-		return;
-
-	drm_kms_helper_poll_fini(priv->drm);
-	drm_atomic_helper_shutdown(priv->drm);
+	return dev->of_node == data;
 }
 
 /* Possible connectors nodes to ignore */
@@ -530,7 +527,6 @@ static const struct dev_pm_ops meson_drv_pm_ops = {
 
 static struct platform_driver meson_drm_platform_driver = {
 	.probe      = meson_drv_probe,
-	.shutdown   = meson_drv_shutdown,
 	.driver     = {
 		.name	= "meson-drm",
 		.of_match_table = dt_match,
-- 
2.37.3

